ppafm/io.py

```diff
@@ -415,9 +416,56 @@ def loadGeometry(fname=None,format=None,params=None):
         if qs is None:
             qs = np.zeros(len(Zs))
         atoms = [list(Zs), list(xyzs[:, 0]), list(xyzs[:, 1]), list(xyzs[:, 2]), list(qs)]
-        nDim = params['gridN'].copy()
+        nDim = params["gridN"].copy()
 
-    return atoms,nDim,lvec
+    # Make sure lvec is a 4x3 array. Use default grid parameters if needed
+    if (lvec is None) or (len(lvec) == 0):
+        lvec = np.zeros((4, 3))
+        lvec[1, :] = params["gridA"].copy()
+        lvec[2, :] = params["gridB"].copy()
+        lvec[3, :] = params["gridC"].copy()
+    else:
+        if len(lvec) == 1:
+            lvec.append(params["gridA"].copy())
+        if len(lvec) == 2:
+            lvec.append(params["gridB"].copy())
+        if len(lvec) == 3:
+            lvec.append(params["gridC"].copy())
+    lvec = np.array(lvec)
+
+    # Shift scanning coordinates to such that actually make sense (being directly comparable to coordinates of atoms)
+    probeMin = params["scanMin"].copy()
+    probeMax = params["scanMax"].copy()
```

Comment:
A minor thing, but I think we want to eventually move toward the PEP8 standard for naming conventions, where variable names are in snake_case instead of camelCase, so here it would be `probe_min` instead of `probeMin`.

Reply:
Corrected